tasks/compile-translations-po-to-json.js

```diff
@@ -11,6 +11,56 @@ function isDirectory(path) {
     }
 }
 
+/*
+
+It iterates all user supplied translations and removes placeholders found in the original string.
+If there are any additional placeholders left, the translation is considered invalid. It is whitespace sensitive.
+
+For example: `{"Update {{sequence}}": "Mettre à jour {{séquence}}"}`
+
+The original placeholder is `{{sequence}}`. It would be removed from the translation, but it doesn't contain it.
+An additional placeholder that was not in the original string is found - `{{séquence}}` and because of this
+the translation is considered invalid and will not be outputted to JSON.
+
+*/
+
+function removeInvalidTranslations(grunt, jsonFilePath, filename) {
+    var translations = JSON.parse(fs.readFileSync(jsonFilePath, 'utf8'));
+
+    const validTranslations = Object.keys(translations).filter((key) => {
+        if (key === '') { // metadata object added by gettext.js
+            return true;
+        }
+
+        const placeHolders = key.match(/{{.+?}}/g);
```

Comment:
just wondering what is that `?` supposed to do?

Reply:
> a {{b}} c {{d}} e

`?` makes it stop at the first occurrence of the closing brackets(if you have multiple placeholders at the same line). If you use `?`, you'll get 2 matches - `{{b}}` and `{{c}}`. If not, you'd get a single match `{{b}} c {{d}}` which would also include text between the placeholders that shouldn't be included.

Reply:
I see that's non greedy switch, thx